agent/internal/agent.go

```diff
@@ -120,51 +118,64 @@ func (a *Agent) run(ctx context.Context) error {
 		return fmt.Errorf("failed to detect devices: %v", devices)
 	}
 
-	a.log.Tracef("setting up %s runtime", a.opts.ContainerRuntime)
-	var cruntime container.ContainerRuntime
-	switch a.opts.ContainerRuntime {
-	case options.PodmanContainerRuntime:
-		acl, sErr := podman.New(a.opts)
-		if sErr != nil {
-			return fmt.Errorf("failed to build podman client: %w", sErr)
-		}
-		defer func() {
-			if cErr := acl.Close(); cErr != nil {
-				a.log.WithError(cErr).Error("failed to close podman client")
-			}
-		}()
-		cruntime = acl
-	case options.ApptainerContainerRuntime:
-		fallthrough
-	case options.SingularityContainerRuntime:
-		acl, sErr := singularity.New(a.opts)
-		if sErr != nil {
-			return fmt.Errorf("failed to build singularity client: %w", sErr)
-		}
-		defer func() {
-			if cErr := acl.Close(); cErr != nil {
-				a.log.WithError(cErr).Error("failed to close singularity client")
-			}
-		}()
-		cruntime = acl
-	case options.DockerContainerRuntime:
-		dcl, dErr := dclient.NewClientWithOpts(dclient.WithAPIVersionNegotiation(), dclient.FromEnv)
-		if dErr != nil {
-			return fmt.Errorf("failed to build docker client: %w", dErr)
-		}
-		defer func() {
-			a.log.Trace("cleaning up docker client")
-			if cErr := dcl.Close(); cErr != nil {
-				a.log.WithError(cErr).Error("failed to close docker client")
-			}
-		}()
-		cl := docker.NewClient(dcl)
-		cruntime = cl
+	a.log.Trace("setting up docker client")
+	dcl, dErr := dclient.NewClientWithOpts(dclient.WithAPIVersionNegotiation(), dclient.FromEnv)
+	if dErr != nil {
+		return fmt.Errorf("failed to build docker client: %w", dErr)
 	}
-
-	a.log.Trace("setting up container manager")
+	defer func() {
+		a.log.Trace("cleaning up docker client")
+		if cErr := dcl.Close(); cErr != nil {
+			a.log.WithError(cErr).Error("failed to close docker client")
+		}
+	}()
+	docker := docker.NewClient(dcl)
+
+	// a.log.Tracef("setting up %s runtime", a.opts.ContainerRuntime)
```

Comment:
Commented a few parts of code instead of completely removing it as we are unclear if we want to deprecate whole of container_runtime (configuration option/variable) or just overloaded use of "container runtime" (container "providers", e.g. docker, podman, apptainer).

I will make corrections as needed in future commits

Reply:
I think the only valid option for container_runtime should be docker

so if I had a config file that specified ```container_runtime: docker``` I would expect that to not error after this change

Reply:
every other container runtime I would expect the agent to return an error message saying it can't run